servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/PublisherFlatMapSingle.java

```diff
@@ -320,14 +300,9 @@ private void doDrainPostProcessing(final long drainCount) {
             }
         }
 
-        private void enqueueAndDrainFailed(Object item) {
-            QueueFullException exception = new QueueFullException("pending");
-            if (item instanceof TerminalNotification) {
-                LOGGER.error("Queue should be unbounded, but an offer failed!", exception);
-                throw exception;
-            } else {
-                onError0(exception, true, true);
-            }
+        private void enqueueFailed(Object item) {
+            LOGGER.error("Queue should be unbounded, but an offer failed for item {}!", item);
+            throw new QueueFullException("pending");
```

Comment:
We discussed [here](https://github.com/apple/servicetalk/pull/1326#discussion_r560579593) that we can't throw inside terminal methods. But this one may potentially throw (if the queue changes). Consider using try-catch where this method is invoked.